pywbem/_logging.py

```diff
@@ -0,0 +1,402 @@
+                if comp != 'all':
+                    cls.create_logger(comp, log_dest=log_dest,
+                                      log_filename=log_filename,
+                                      log_detail_level=log_detail_level,
+                                      log_level=log_level)
+
+        # Otherwise process results of any recursive calls above
+        else:
+            if log_dest == 'stderr':
+                handler = logging.StreamHandler()
+                format_string = '%(asctime)s-%(name)s-%(message)s'
+            elif log_dest == 'file':
+                if not log_filename:
+                    raise ValueError('Filename required if log destination '
+                                     'is "file"')
+                # pylint: disable=redefined-variable-type
+                handler = logging.FileHandler(log_filename)
+                format_string = '%(asctime)s-%(name)s-%(message)s'
+            else:
+                assert(log_dest == 'none')
+                handler = None
+                format_string = None
+
+            # set the logger name based on the log_component.
+            if log_component == 'http':
+                logger_name = LOG_HTTP_NAME
+                if log_level is None:
+                    log_level = 'debug'
+            elif log_component == 'ops':
+                logger_name = LOG_OPS_CALLS_NAME
+                if log_level is None:
+                    log_level = 'debug'
+            else:
+                raise ValueError('Invalid log_component %s' % log_component)
+
+            # Set the log level based on the log_level input
+            level = getattr(logging, log_level.upper(), None)
+            # check logging log_level_choices have not changed from expected
+            assert isinstance(level, int)
+            if level is None:
+                raise ValueError('Invalid log level %s specified. Must be one '
+                                 'of %s.' % (log_level, LOG_LEVELS))
+
+            # create named logger
+            if handler:
+                handler.setFormatter(logging.Formatter(format_string))
+                logger = logging.getLogger(logger_name)
+                logger.addHandler(handler)
+                logger.setLevel(level)
+
+            # save the detail level in the dict that is part of this class.
+            # All members of this tuple are just viewing except detail
+            # that is used by individual loggers
+            cls.loggers[logger_name] = (log_detail_level, log_level,
+                                        log_dest, log_filename)
+
+    @classmethod
+    def get_logger(cls, logger_name):
+        """
+            Get the name of a logger that has been defined.
+        """
+        if logger_name in cls.loggers:
+            return cls.loggers[logger_name][0]
+        return None
+
+    @classmethod
+    def get_logger_info(cls, logger_name):
+        """Get information about a logger by name"""
```

Comment:
Add documentation about the return value. Specify the input parameter in the documentation.